README.md

```diff
@@ -145,73 +145,64 @@ in the Minishift docker environment, as detailed in next section.
-This can be done with the following command:
+### Deploy to openshift.io 
 
 ```bash
-minishift start
+SCRIPT_URL=https://raw.githubusercontent.com/redhat-developer/rh-che/master/dev-scripts/openshift_deploy.sh
+export OPENSHIFT_FLAVOR=osio && curl -fsSL ${SCRIPT_URL} -o get-che.sh && sh get-che.sh
 ```
 
-#### Deploy to Minishift 
+### Deployment Options
 
-```bash
-dev-scripts/minishift_deploy.sh
-```
-By default, the docker image of the deployed to Minishift is the last built `nightly-fabric8` image.
+You can set different deployment options using environment variables:
 
-However, you can override this by providing the `CHE_IMAGE_TAG` variable:
+* `OPENSHIFT_FLAVOR`: possible values are `minishift` and `osio` (default is `minishift`)
+* `OPENSHIFT_ENDPOINT`: url of the OpenShift API (default is `https://$(minishift ip):8443/` for minishift, `https://api.starter-us-east-2.openshift.com` for osio)
+* `OPENSHIFT_TOKEN` (default is unset)
+* `CHE_OPENSHIFT_PROJECT`: the OpenShift namespace where Che will be deployed (default is `eclipse-che` for minishift and `${OPENSHIFT_ID}-che` for osio)
+* `CHE_IMAGE_REPO`: `che-server` Docker image repository that will be used for deployment (default is `docker.io/rhchestage/che-server`)
+* `CHE_IMAGE_TAG`: `che-server` Docker image tag that will be used for deployment (default is `nightly-fabric8`)
+* `CHE_LOG_LEVEL`: Log level of che-server (default is `DEBUG`)
+* `CHE_DEBUGGING_ENABLED`: If set to `true` the script will create the OpenShift service to debug che-server (default is `true`)
+* `CHE_KEYCLOAK_DISABLED`: If this is set to true Keycloack authentication will be disabled (default is `true` for minishift, `false` for osio)
 
-```bash
-bash -c "export CHE_IMAGE_TAG=nightly ; dev-scripts/minishift_deploy.sh"
-```
+#### minishift only options
 
-You can even override the name of the docker image with the `CHE_IMAGE_REPO` variable:
+* `OPENSHIFT_USERNAME`: username to login on the OpenShift cluster. Ignored if `OPENSHIFT_TOKEN` is set (default is `developer`)
+* `OPENSHIFT_PASSWORD`: password to login on the OpenShift cluster. Ignored if `OPENSHIFT_TOKEN` is set (default is `developer`)
 
-```bash
-bash -c "export CHE_IMAGE_REPO=rhche/che-server ; export CHE_IMAGE_TAG=nightly ; dev-scripts/minishift_deploy.sh"
-```
-
-By default, Che is deployed on minishift with the Keycloak integration *disabled*.
-
-However, to explicitly enable the Keycloak integration, you can override the `CHE_KEYCLOAK_DISABLED` variable:
+#### openshift.io only options
 
-```bash
-bash -c "export CHE_KEYCLOAK_DISABLED=false ; dev-scripts/minishift_deploy.sh"
-```
+* `OPENSHIFT_ID`: Red Hat Developers username with some exceptions:
+  * If the username is an e-mail address the OPENSHIFT_ID will the part before the @ symbol (e.g. for cbrown@redhat.com `OPENSHIFT_ID` will be cbrown)
+  * Dots should be replaced by dashes ((e.g. for c.brown `OPENSHIFT_ID` will be c-brown)
```

Comment:
Not reliably, these are just guidelines. I'd rather have nothing than something that doesn't work 100% of the time. 

The username generation has to be done to avoid conflicts, and since we're using email addresses, it'll be unpredictable when e.g. two users sign up with `email@domain1.com` and `email@domain2.com`.